### PR TITLE
EDSC-2192: Save Projects

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: static/src/js/actions/savedProject.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  } = savedProject
+
+  const { location } = router
+  const { pathname, search } = location
+
+  // If there isn't a path saved yet, get it from the URL
+  let realPath = path
+  if (!path) realPath = pathname + search
+
+  const requestObject = new ProjectRequest()
```
[the remark this answers is not in the document]
Comment (macrouch) [reply static/src/js/actions/savedProject.js]: Saving projects doesn't require the user to be logged in, if the user is logged in I pass the authToken in as a parameter in the body

File: static/src/js/actions/savedProject.js [hunk 2]
```diff
@@ -2,19 +2,24 @@ import { replace } from 'connected-react-router'
...
+import { removeSavedProject } from './savedProjects'
 
 export const updateSavedProject = payload => ({
   type: UPDATE_SAVED_PROJECT,
   payload
 })
 
+/**
+ * Action called when the project name is being updated
+ * @param {*} name New project name
```
Comment (abbottry) [inline static/src/js/actions/savedProject.js]: String

File: static/src/js/actions/savedProject.js [hunk 3]
```diff
@@ -58,18 +63,26 @@ export const updateProjectName = name => (dispatch, getState) => {
   return response
 }
 
+/**
+ * Action called when deleting a saved project
+ * @param {*} projectId Project Id to delete
```
Comment (abbottry) [inline static/src/js/actions/savedProject.js]: String (I think)

File: static/src/js/components/SavedProjects/SavedProjects.js [hunk 4]
```diff
@@ -19,11 +19,20 @@ export class SavedProjects extends Component {
     this.handleDeleteSavedProject = this.handleDeleteSavedProject.bind(this)
   }
 
+  /**
+   * Returns the URL for a project
+   * @param {*} path Path field saved in the project
```
Comment (abbottry) [inline static/src/js/components/SavedProjects/SavedProjects.js]: String

File: static/src/js/components/SavedProjects/SavedProjects.js [hunk 5]
```diff
@@ -19,11 +19,20 @@ export class SavedProjects extends Component {
...
+   * @param {*} id The project Id
```
Comment (abbottry) [inline static/src/js/components/SavedProjects/SavedProjects.js]: String (I think)

File: static/src/js/components/SavedProjects/SavedProjects.js [hunk 6]
```diff
@@ -19,11 +19,20 @@ export class SavedProjects extends Component {
...
+   */
   projectTo(path, id) {
     const [pathname] = path.split('?')
     return `${pathname}?projectId=${id}`
   }
 
+  /**
+   * Determines the number of collections saved in the project path
+   * @param {*} path Project path
```
Comment (abbottry) [inline static/src/js/components/SavedProjects/SavedProjects.js]: String

**Pull request merged**